addons/account/models/ir_ui_menu.py

```diff
@@ -6,6 +6,7 @@ class IrUiMenu(models.Model):
 
     def _load_menus_blacklist(self):
         res = super()._load_menus_blacklist()
-        if not any(company.check_account_audit_trail for company in self.env.user.company_ids):
-            res.append(self.env.ref('account.account_audit_trail_menu').id)
+        menu = self.env.ref('account.account_audit_trail_menu', raise_if_not_found=False)
+        if not any(company.check_account_audit_trail for company in self.env.user.company_ids) and menu:
```

Comment:
It's a nitpick - but I have done that :)
```suggestion
        if menu and not any(company.check_account_audit_trail for company in self.env.user.company_ids):
```